### DISCUSSION TITLE
how to install and run HammerDB on ARM linux

### DISCUSSION THREAD

**Discussion by ryuxin** [TPC-Council/HammerDB#282] [Q&A] [answered]
I only found HammerDB  binary for x86.
how to install and run HammerDB on ARM linux?

Thanks

**Comment by sm-shaw** [comment, marked as answer]
Yes, at the moment binaries are only built for x86-64. However, HammerDB on x86-64 can connect to a database on any supported platform, the link below is an example from PostgreSQL on IBM Z - therefore to test databases on ARM Linux this is also by far the easiest and quickest way to do it. 

[PostgreSQL: Experiences and tuning recommendations on Linux on IBM Z](https://developer.ibm.com/tutorials/postgresql-experiences-tuning-recomendations-linux-on-ibm-z/)
![highlevel](https://user-images.githubusercontent.com/38044085/140151676-9fb33c31-3136-4217-8c8e-1fa9f1511499.PNG)

As more detailed background, there are no current plans to build HammerDB for ARM. The HammerDB code on GitHub is the same for both Linux and Windows, the difference in the binary releases is in the bin and lib directories and this is where the compiled 3rd party libraries for a particular platform reside. Therefore, to build HammerDB for non-x86-64 platforms would require compiling [Tcl/Tk](https://www.tcl.tk/software/tcltk/) for that platform as well as the additional libraries - everything is open source including any changes HammerDB has made, for example Db2tcl here https://github.com/memmertoIBM/db2tcl.  However, 3 of the supported databases Oracle, SQL Server and Db2 do not currently have libraries for Linux ARM and therefore that would rule out a lot of the existing functionality and user base. Where code does compile, it would also need testing for performance, stability and compatibility. 

Additionally, there is the consideration of development platforms and ecosystem such as tools that are the same architecture as the servers that the database runs on. Linux and Windows x86-64 laptops and desktops are widely available and customizable for database development, high memory, SSD disks and UHD displays e.g. I use a laptop from [entroware](https://www.entroware.com/store/) for Linux.  Currently, for ARM (and other non-x86-64 platforms) the support for development environments from laptop, desktop through to server are not as available. 

Therefore,  (also adding the need for ongoing support and maintenance) the Linux ARM ecosystem is not developed to the extent that the anticipated time and cost invested in building, testing and supporting the platform would match the expected downloads compared to time and cost spent in other development areas on both Linux and Windows.  This is not to say that if someone had the interest, development systems and time to do it, it could be a project to see if the software packages used can be built on any particular platform they wish to use. However, as above, there are no current plans from the core team to do this and following the diagram above is the easiest way to test databases on non-x86-64 platforms at the present time.

**Comment by sm-shaw** [comment]
Issue moved to discussion topic

**Comment by Tank-Tong** [comment]
@ryuxin Have you get the method install HammerDB on ARM?

**Reply by abondvt89** [reply]
The original answer still applies.  HammerDB itself doesn't run on the ARM architecture, but it can connect to databases running on ARM machines.  Right now there are no plans to release a HammerDB version that runs on ARM.  However, if you would like to contribute the development and testing to make it work on ARM we would be happy to work with you on integrating such an effort.
